### ISSUE TITLE
Using computed property for searching

### ISSUE THREAD

**Issue by lexa7979** [hitchyjs/plugin-odem#14]
I defined a model *jobs* with a property *state* which can be `created`, `failed`, `prepared`, `printing`, ... Additionally, the model has a computed *spoolAction* which shall be set or unset depending on the current state:

```
spoolAction() {
	switch ( this.state ) {
	case "printing" :
		return "print";
	case "print-cancelling" :
		return "cancel";
	case "deleting" :
		return "delete";
	default :
		return null;
	}
}
```

**Problem**

I unfortunately can't use the computed *spoolAction* for a search on all instances of *jobs*:
```
this.api.runtime.models.Jobs.find( { notnull: { name: "spoolAction" } } )
```

**Observations**

I always get the error **"no such property: spoolAction"**.

**Tested solution**

The error disappears if I define the model *jobs* with an index:
```
indices: {
    spoolAction: { propertyType: "string" }
},
```

**New problems**

It looks like hitchy-plugin-odem-rest get's problems with the new index. A request to `GET /api/jobs?count=1&limit=1` now always delivers a empty list of items, even after inserting new jobs which still can be fetched with `GET /api/jobs/<uuid>`.

**Comment by lexa7979** [comment]
I'm unsure if this issue is related to hitchy-plugin-odem or hitchy-plugin-odem-rest!? Maybe it belongs to both... ;-)

**Comment by soletan** [comment]
The non-indexed comparison tester didn't obey computed properties but was limited to working with actual properties, only. The follow-up issue was due to incompletely refactoring some previously existing typo in one of the internal APIs.

Either issue has been fixed in v0.2.9 available now.

**Comment by lexa7979** [comment]
It's working now - thanks.
